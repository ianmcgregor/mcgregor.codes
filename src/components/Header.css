:root {}
.Header {
    align-items: center;
    border-bottom: solid var(--border-color);
    border-width: var(--border-width);
    background-color: var(--header-bg-color);
    display: flex;
    flex-shrink: 0;
    justify-content: space-between;
    padding: var(--spacing-wide);
    position: relative;
    padding-bottom: 22px;
}
.Header-title {
    color: var(--header-title-color);
    flex: 1;
    font-size: var(--font-size-medium);
    font-weight: normal;
    line-height: 1;
    margin: 0;
}
@media (--narrow) {
    .Header-title {
        font-size: var(--font-size-x-large);
    }
}
.Header-menu {
    display: flex;
}
.Header-menuItem {
    display: block;
    margin: 0 var(--spacing-narrow);
}
@media (--narrow) {
    .Header-menuItem {
        margin: 0 var(--spacing);
    }
}
@media (--medium) {
    .Header-menuItem {
        margin: 0 var(--spacing-wide);
    }
}
.Header-menuItem:last-child {
    margin-right: 0;
}
.Header-link {
    color: var(--header-link-color);
    font-size: var(--font-size-small);
}
@media (--medium) {
    .Header-link {
        font-size: var(--font-size-large);
    }
}
.Header-link::before {
    background-color: var(--header-link-hover-color);
}
