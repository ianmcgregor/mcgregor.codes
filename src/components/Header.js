import React from 'react';
import {Link, IndexLink} from 'react-router';

class Header extends React.Component {

    render () {

        const {title, pages} = this.props;

        return (
            <header className="Header">
                <IndexLink to="/" className="Header-titleLink" activeClassName="is-active">
                    <h1 className="Header-title">
                        {title}
                    </h1>
                </IndexLink>
                <nav>
                    <ul className="Header-menu">
                        {pages.map((page) => {
                            return (
                                <li className="Header-menuItem">
                                    <Link to={page.route} className="Header-link u-link" activeClassName="is-active">
                                        {page.title}
                                    </Link>
                                </li>
                            );
                        })}
                    </ul>
                </nav>
            </header>
        );
    }
}

export {Header as default};
