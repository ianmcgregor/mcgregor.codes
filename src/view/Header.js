import React from 'react';
// import Boids from '../components/Boids';
import Link from '../components/Link';

export default function Header(props) {
    const {sections, currentSection} = props;

    return (
        <header className="Header u-padH" data-path="">
            <Link path="/" currentSection={currentSection} className="Header-titleLink" activeClassName="is-active">
                <h1 className="Header-title">
                    M<sup>c</sup>Gregor Codes
                </h1>
            </Link>
            <nav className="Header-nav">
                <ul className="Header-menu">
                    {sections.map((section, i) => {
                        return (
                            <li className="Header-menuItem" key={'Header-menuItem' + i}>
                                <Link
                                    path={`/${section.slug}/`}
                                    currentSection={currentSection}
                                    className="Header-link u-link"
                                    activeClassName="is-active">
                                        {section.title}
                                </Link>
                            </li>
                        );
                    })}
                </ul>
            </nav>
            <p className="Header-subtitle">
                Specialist in creative, experiential
                web sites, games and applications.
            </p>
            {/* <Boids /> */}
        </header>
    );
}
