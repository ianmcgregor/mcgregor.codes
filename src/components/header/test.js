import React from 'react';
import {render} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import Header from './';

const props = {
    links: [
        {
            label: 'Work',
            url: '/'
        },
        {
            label: 'About',
            url: '/about/'
        }
    ]
};

describe('Header component', () => {
    it('should render', () => {
        const el = render((
            <MemoryRouter>
                <Header {...props}/>
            </MemoryRouter>
        ));
        expect(el).toMatchSnapshot();
    });
});
