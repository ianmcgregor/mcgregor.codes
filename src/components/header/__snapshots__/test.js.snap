// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-dnqmqq etZNfp",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-iwsKbI vffzw",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-ifAKCX hbKAms",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-EHOje cyvMVa",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-bZQynM gAjREB",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "sc-gzVnrw fbHcEw",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "aria-current": "true",
                              "class": "sc-htoDjs biXjnK active",
                              "href": "/",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "sc-bdVaJa fGHDiW",
                                },
                                "children": Array [
                                  Object {
                                    "attribs": Object {
                                      "class": "sc-gZMcBi dyJVgy",
                                    },
                                    "children": Array [
                                      Object {
                                        "attribs": Object {
                                          "class": "sc-htpNat eIqxAw",
                                        },
                                        "children": Array [
                                          Object {
                                            "attribs": Object {
                                              "class": "sc-bxivhb jZLiij",
                                            },
                                            "children": Array [
                                              Object {
                                                "data": "Work",
                                                "next": null,
                                                "parent": [Circular],
                                                "prev": null,
                                                "type": "text",
                                              },
                                            ],
                                            "name": "span",
                                            "namespace": "http://www.w3.org/1999/xhtml",
                                            "next": null,
                                            "parent": [Circular],
                                            "prev": null,
                                            "type": "tag",
                                            "x-attribsNamespace": Object {
                                              "class": undefined,
                                            },
                                            "x-attribsPrefix": Object {
                                              "class": undefined,
                                            },
                                          },
                                        ],
                                        "name": "span",
                                        "namespace": "http://www.w3.org/1999/xhtml",
                                        "next": null,
                                        "parent": [Circular],
                                        "prev": null,
                                        "type": "tag",
                                        "x-attribsNamespace": Object {
                                          "class": undefined,
                                        },
                                        "x-attribsPrefix": Object {
                                          "class": undefined,
                                        },
                                      },
                                    ],
                                    "name": "span",
                                    "namespace": "http://www.w3.org/1999/xhtml",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                    "x-attribsNamespace": Object {
                                      "class": undefined,
                                    },
                                    "x-attribsPrefix": Object {
                                      "class": undefined,
                                    },
                                  },
                                ],
                                "name": "div",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                },
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "aria-current": undefined,
                              "class": undefined,
                              "href": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "aria-current": undefined,
                              "class": undefined,
                              "href": undefined,
                            },
                          },
                        ],
                        "name": "li",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "class": "sc-gzVnrw fbHcEw",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "aria-current": "false",
                                "class": "sc-htoDjs biXjnK",
                                "href": "/about/",
                              },
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "class": "sc-bdVaJa fGHDiW",
                                  },
                                  "children": Array [
                                    Object {
                                      "attribs": Object {
                                        "class": "sc-gqjmRU fAuRiS",
                                      },
                                      "children": Array [
                                        Object {
                                          "attribs": Object {
                                            "class": "sc-htpNat eIqxAw",
                                          },
                                          "children": Array [
                                            Object {
                                              "attribs": Object {
                                                "class": "sc-bxivhb jZLiij",
                                              },
                                              "children": Array [
                                                Object {
                                                  "data": "About",
                                                  "next": null,
                                                  "parent": [Circular],
                                                  "prev": null,
                                                  "type": "text",
                                                },
                                              ],
                                              "name": "span",
                                              "namespace": "http://www.w3.org/1999/xhtml",
                                              "next": null,
                                              "parent": [Circular],
                                              "prev": null,
                                              "type": "tag",
                                              "x-attribsNamespace": Object {
                                                "class": undefined,
                                              },
                                              "x-attribsPrefix": Object {
                                                "class": undefined,
                                              },
                                            },
                                          ],
                                          "name": "span",
                                          "namespace": "http://www.w3.org/1999/xhtml",
                                          "next": null,
                                          "parent": [Circular],
                                          "prev": null,
                                          "type": "tag",
                                          "x-attribsNamespace": Object {
                                            "class": undefined,
                                          },
                                          "x-attribsPrefix": Object {
                                            "class": undefined,
                                          },
                                        },
                                      ],
                                      "name": "span",
                                      "namespace": "http://www.w3.org/1999/xhtml",
                                      "next": null,
                                      "parent": [Circular],
                                      "prev": null,
                                      "type": "tag",
                                      "x-attribsNamespace": Object {
                                        "class": undefined,
                                      },
                                      "x-attribsPrefix": Object {
                                        "class": undefined,
                                      },
                                    },
                                  ],
                                  "name": "div",
                                  "namespace": "http://www.w3.org/1999/xhtml",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                  "x-attribsNamespace": Object {
                                    "class": undefined,
                                  },
                                  "x-attribsPrefix": Object {
                                    "class": undefined,
                                  },
                                },
                              ],
                              "name": "a",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "aria-current": undefined,
                                "class": undefined,
                                "href": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "aria-current": undefined,
                                "class": undefined,
                                "href": undefined,
                              },
                            },
                          ],
                          "name": "li",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                      Object {
                        "attribs": Object {
                          "class": "sc-gzVnrw fbHcEw",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "aria-current": "false",
                              "class": "sc-htoDjs biXjnK",
                              "href": "/about/",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "sc-bdVaJa fGHDiW",
                                },
                                "children": Array [
                                  Object {
                                    "attribs": Object {
                                      "class": "sc-gqjmRU fAuRiS",
                                    },
                                    "children": Array [
                                      Object {
                                        "attribs": Object {
                                          "class": "sc-htpNat eIqxAw",
                                        },
                                        "children": Array [
                                          Object {
                                            "attribs": Object {
                                              "class": "sc-bxivhb jZLiij",
                                            },
                                            "children": Array [
                                              Object {
                                                "data": "About",
                                                "next": null,
                                                "parent": [Circular],
                                                "prev": null,
                                                "type": "text",
                                              },
                                            ],
                                            "name": "span",
                                            "namespace": "http://www.w3.org/1999/xhtml",
                                            "next": null,
                                            "parent": [Circular],
                                            "prev": null,
                                            "type": "tag",
                                            "x-attribsNamespace": Object {
                                              "class": undefined,
                                            },
                                            "x-attribsPrefix": Object {
                                              "class": undefined,
                                            },
                                          },
                                        ],
                                        "name": "span",
                                        "namespace": "http://www.w3.org/1999/xhtml",
                                        "next": null,
                                        "parent": [Circular],
                                        "prev": null,
                                        "type": "tag",
                                        "x-attribsNamespace": Object {
                                          "class": undefined,
                                        },
                                        "x-attribsPrefix": Object {
                                          "class": undefined,
                                        },
                                      },
                                    ],
                                    "name": "span",
                                    "namespace": "http://www.w3.org/1999/xhtml",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                    "x-attribsNamespace": Object {
                                      "class": undefined,
                                    },
                                    "x-attribsPrefix": Object {
                                      "class": undefined,
                                    },
                                  },
                                ],
                                "name": "div",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                },
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "aria-current": undefined,
                              "class": undefined,
                              "href": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "aria-current": undefined,
                              "class": undefined,
                              "href": undefined,
                            },
                          },
                        ],
                        "name": "li",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "class": "sc-gzVnrw fbHcEw",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "aria-current": "true",
                                "class": "sc-htoDjs biXjnK active",
                                "href": "/",
                              },
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "class": "sc-bdVaJa fGHDiW",
                                  },
                                  "children": Array [
                                    Object {
                                      "attribs": Object {
                                        "class": "sc-gZMcBi dyJVgy",
                                      },
                                      "children": Array [
                                        Object {
                                          "attribs": Object {
                                            "class": "sc-htpNat eIqxAw",
                                          },
                                          "children": Array [
                                            Object {
                                              "attribs": Object {
                                                "class": "sc-bxivhb jZLiij",
                                              },
                                              "children": Array [
                                                Object {
                                                  "data": "Work",
                                                  "next": null,
                                                  "parent": [Circular],
                                                  "prev": null,
                                                  "type": "text",
                                                },
                                              ],
                                              "name": "span",
                                              "namespace": "http://www.w3.org/1999/xhtml",
                                              "next": null,
                                              "parent": [Circular],
                                              "prev": null,
                                              "type": "tag",
                                              "x-attribsNamespace": Object {
                                                "class": undefined,
                                              },
                                              "x-attribsPrefix": Object {
                                                "class": undefined,
                                              },
                                            },
                                          ],
                                          "name": "span",
                                          "namespace": "http://www.w3.org/1999/xhtml",
                                          "next": null,
                                          "parent": [Circular],
                                          "prev": null,
                                          "type": "tag",
                                          "x-attribsNamespace": Object {
                                            "class": undefined,
                                          },
                                          "x-attribsPrefix": Object {
                                            "class": undefined,
                                          },
                                        },
                                      ],
                                      "name": "span",
                                      "namespace": "http://www.w3.org/1999/xhtml",
                                      "next": null,
                                      "parent": [Circular],
                                      "prev": null,
                                      "type": "tag",
                                      "x-attribsNamespace": Object {
                                        "class": undefined,
                                      },
                                      "x-attribsPrefix": Object {
                                        "class": undefined,
                                      },
                                    },
                                  ],
                                  "name": "div",
                                  "namespace": "http://www.w3.org/1999/xhtml",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                  "x-attribsNamespace": Object {
                                    "class": undefined,
                                  },
                                  "x-attribsPrefix": Object {
                                    "class": undefined,
                                  },
                                },
                              ],
                              "name": "a",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "aria-current": undefined,
                                "class": undefined,
                                "href": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "aria-current": undefined,
                                "class": undefined,
                                "href": undefined,
                              },
                            },
                          ],
                          "name": "li",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "ul",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "nav",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "header",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
